src/main/java/com/pnu/ordermanagementapp/client/ClientDbAdapter.java

```diff
@@ -0,0 +1,45 @@
+package com.pnu.ordermanagementapp.client;
+
+import com.pnu.ordermanagementapp.adapter.DbAdapter;
+import com.pnu.ordermanagementapp.model.Client;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class ClientDbAdapter implements DbAdapter<Client> {
+    private ClientRepository repository;
+
+    @Autowired
+    public ClientDbAdapter(ClientRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public List<Client> findAll() {
+        return (List<Client>) repository.findAll();
+    }
+
+    @Override
+    public void create(Client obj) {
+        repository.save(obj);
+    }
+
+    @Override
+    public void update(Client obj) {
+        findClientByIdOrThrowException(obj.getId());
+        repository.save(obj);
+    }
+
+    @Override
+    public void delete(Long id) {
+        Client client = findClientByIdOrThrowException(id);
+        repository.delete(client);
+    }
+
+    private Client findClientByIdOrThrowException(Long id){
+        return repository.findById(id)
```

Comment:
will it bring user to some custom page or the user will get full stack trace in browser?
I'd suggest just skip deleting, if there is no client, with provided id for now, later we will handle edge cases somehow

Reply:
I've implemented an error page!